scijava/scijava-ops-engine/src/main/java/org/scijava/ops/engine/matcher/simplify/SimplifiedOpInfo.java

```diff
@@ -294,29 +226,51 @@ public String version() {
 	public String id() {
 		// original Op
 		StringBuilder sb = new StringBuilder(IMPL_DECLARATION);
-		// input simplifiers
-		for (InfoTree i : metadata.inputSimplifierChains()) {
-			sb.append(INPUT_SIMPLIFIER_DELIMITER);
-			sb.append(i.signature());
-		}
 		// input focusers
-		for (InfoTree i : metadata.inputFocuserChains()) {
+		for (RichOp<Function<?, ?>> i : inputFocusers) {
 			sb.append(INPUT_FOCUSER_DELIMITER);
-			sb.append(i.signature());
+			sb.append(i.infoTree().signature());
 		}
 		// output simplifier
 		sb.append(OUTPUT_SIMPLIFIER_DELIMITER);
-		sb.append(metadata.outputSimplifierChain().signature());
-		// output focuser
-		sb.append(OUTPUT_FOCUSER_DELIMITER);
-		sb.append(metadata.outputFocuserChain().signature());
-
+		sb.append(outputSimplifier.infoTree().signature());
 		// output copier
-		sb.append(OUTPUT_COPIER_DELIMITER);
-		sb.append(metadata.copyOpChain().signature());
+		if (copyOp != null) {
+			sb.append(OUTPUT_COPIER_DELIMITER);
+			sb.append(copyOp.infoTree().signature());
+		}
 		// original info
 		sb.append(ORIGINAL_INFO);
 		sb.append(srcInfo().id());
 		return sb.toString();
 	}
+
+	private Type[] inTypes(Type[] originalInputs, List<RichOp<Function<?, ?>>> inputFocusers) {
+		Map<TypeVariable<?>, Type> typeAssigns = new HashMap<>();
+		Type[] inTypes = new Type[originalInputs.length];
+		// TODO: This feels kind of inefficient, but we have to do each individually
```

Comment:
Should this be an `NB:` prefix instead of `TODO:`? It is just explaining why the code is written this way...

Reply:
Sure, changed!